internal/venice-common/src/main/java/com/linkedin/venice/utils/DictionaryUtils.java

```diff
@@ -70,6 +70,11 @@ public static ByteBuffer readDictionaryFromKafka(
     while (!startOfPushReceived) {
       Map<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>> messages =
           pubSubConsumer.poll(10 * Time.MS_PER_SECOND);
+
+      if (!messages.containsKey(pubSubTopicPartition)) {
```

Comment:
can we add some delay here?  else it will bombard the kafka poll.

Reply:
`poll(10 * Time.MS_PER_SECOND)` adds the delay inherently because Kafka consumer will wait for 10 sec on every poll. Do we need to add more?